src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -230,7 +230,7 @@ private static byte[][] GetSqlServerSPNs(DataSource dataSource, string serverSPN
             }
             else if (!string.IsNullOrWhiteSpace(dataSource.InstanceName))
             {
-                postfix = dataSource.InstanceName;
+                postfix = (dataSource._connectionProtocol == DataSource.Protocol.TCP ? dataSource.ResolvedPort.ToString() : dataSource.InstanceName);
```

Comment:
I'll leave this up to you, but I think it's easier to read if it's separated since it's different conditional logic.

else if (dataSource._connectionProtocol == DataSource.Protocol.TCP)
    postfix = dataSource.ResolvedPort.ToString()
else if (!string.IsNullOrWhiteSpace(dataSource.InstanceName))
    postfix = dataSource.InstanceName;

Do you also need to check for dataSource.IsSsrpRequired? ResolvedPort  is only assigned when it's true.

Reply:
This is David's code which looks cleaner to me.